src/create-route-map.js

```diff
@@ -43,6 +43,17 @@ function addRouteRecord (
   }
 
   if (route.children) {
+    // Warn if route is named and has a default child route.
+    // If users navigate to this route by name, the default child will
+    // not be rendered (GH Issue #629)
+    if (process.env.NODE_ENV !== 'production') {
+      if (route.name && route.children.some(child => /^\/?$/.test(child.path))) {
```

Comment:
Why should we check for route.children? That's already checked for on line 45.

And the failed tests are transition e2e tests, the unit test (which include this test) run without error on CircleCI